x-pack/plugins/observability/common/custom_threshold_rule/types.ts

```diff
@@ -269,22 +109,22 @@ export enum InfraFormatterType {
   percent = 'percent',
 }
 
-export type SnapshotCustomAggregation = typeof SNAPSHOT_CUSTOM_AGGREGATIONS[number];
-const snapshotCustomAggregationKeys = SNAPSHOT_CUSTOM_AGGREGATIONS.reduce<
-  Record<SnapshotCustomAggregation, null>
->((acc, agg) => ({ ...acc, [agg]: null }), {} as Record<SnapshotCustomAggregation, null>);
-
-export const SnapshotCustomAggregationRT = rt.keyof(snapshotCustomAggregationKeys);
-
-export const SnapshotCustomMetricInputRT = rt.intersection([
-  rt.type({
-    type: rt.literal('custom'),
-    field: rt.string,
-    aggregation: SnapshotCustomAggregationRT,
-    id: rt.string,
-  }),
-  rt.partial({
-    label: rt.string,
-  }),
-]);
-export type SnapshotCustomMetricInput = rt.TypeOf<typeof SnapshotCustomMetricInputRT>;
+/*
+ * Utils
+ *
+ * This utility function can be used to turn a TypeScript enum into a io-ts codec.
+ */
+export function fromEnum<EnumType extends string>(
```

Comment:
What does this function do?

Reply:
This is for turning a Typescript enum into an io-ts validation. I used it for the `Aggregators` enum.

Reply:
there is already some nice utils exposed via perhaps worth reusing that or move that into packages/kbn-io-ts-utils
`import { enumeration } from '@kbn/securitysolution-io-ts-types';
`